### PR TITLE
Add glossary terms: slate, volto-slate and elementEditor

### PR DESCRIPTION
**Pull request by nileshgulia1**
refs: https://github.com/plone/volto/pull/3436#discussion_r903177886

### CODE REVIEW

File: docs/glossary.md [hunk 1]
```diff
@@ -319,7 +319,19 @@ Classic UI
...
+    It enables you to create powerful, intuitive editors similar to those you've probably used in Medium, Dropbox Paper, or Google Docs.
+
+volto-slate
+    `volto-slate` is an interactive default text editor for Volto, developed on top of {term}`Slate` while offering enhanced WYSIWYG functionality and behavior.
+
+elementEditor
+    A generic {term}`volto-slate` plugin architecture that can be used to create other editor interactions that follow the pattern of having a button that toggles a format (an inline element).
+    It also creates a separate edit form for advanced customization of the data attached to the element.
+
+``````
```
Comment (stevepiercy) [inline docs/glossary.md]: Too many backticks.

````suggestion
```
````

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Looks good, with one change, please.

File: docs/glossary.md (hunk 1 above)
Comment (nileshgulia1) [reply docs/glossary.md]: done.Thanks

**Review by stevepiercy — APPROVED** [review]
Thank you!

**Pull request merged**